components/utils/DataProvider.js

```diff
@@ -13,34 +15,87 @@ class DataProvider extends Component {
       networks: ["Testnet One", "Testnet Two"],
       currentNetwork: null,
       updateNetwork: this.updateNetwork,
-      details: dummyData.details,
-      blocks: dummyData.blocks,
+      details: this.props.details,
+      blocks: this.props.blocks,
       transactions: dummyData.transactions
     };
   }
 
-  componentWillMount() {
+  static async getInitialProps(ctx) {
+    const [details, blocks] = await Promise.all([
+      DataProvider.getDetails(),
+      DataProvider.getBlocksInfo()
+    ]);
+    return {
+      details,
+      blocks
+    };
+  }
+
+  componentDidMount() {
     this.updateNetwork(0);
   }
 
+  static async getBlocksInfo() {
+    let blocks;
+    try {
+      blocks = await call(".select", [
+        "SELECT hash, height, timestamp, list_of_approvals as listOfApprovals FROM blocks LIMIT 10"
```

Comment:
@frol What order are the blocks returned in? Are the blocks created later on top?

Reply:
Good point. The DB structure is still lacking, so this was barely an example of how to fetch data from the backends.

Reply:
Ah, that's fine. Maybe you can add an `ORDER_BY height DESC` to the query, which should fix it right?